ccm_web/client/src/utils/validation.ts

```diff
@@ -0,0 +1,46 @@
+import { string, StringSchema, ValidationError } from 'yup'
+
+// Yup: https://github.com/jquense/yup
+
+// Schemas
+
+const createExceededMessage = (fieldName: string, max: number): string => {
+  return `Value for the ${fieldName} must be ${max} characters in length or less.`
+}
+
+const createBlankMessage = (fieldName: string): string => `Value for the ${fieldName} may not be blank.`
+
+const createCanvasEntityNameSchema = (fieldName: string): StringSchema => {
+  return string().required(createBlankMessage(fieldName))
+    .max(255, ({ max }) => createExceededMessage(fieldName, max))
+}
+
+export const courseNameSchema = createCanvasEntityNameSchema('course name')
+export const assignmentHeaderSchema = createCanvasEntityNameSchema('assignment header')
+export const sectionNameSchema = createCanvasEntityNameSchema('section name')
+
+// Type validator(s)
+
+export interface ValidationResult {
+  transformedValue: string | undefined
+  isValid: boolean
+  messages: readonly string[]
+}
+
+export function validateString (value: string | undefined, schema: StringSchema): ValidationResult {
+  let transformedValue: string | undefined
+  let isValid = true
+  let messages: string[] = []
+  try {
+    transformedValue = schema.validateSync(value)
+  } catch (error) {
+    if (!(error instanceof ValidationError)) {
```

Comment:
Can you tell me a case which trigger this?

Reply:
Actually, no, this shouldn't happen, but because of TypeScript, I do have to handle it. If it does happen, we'll see an error in the logs.

Reply:
Yeah... I have done that before. In order to make typescript happy i need to add checks.